backend/convert_to_protobuf.go

```diff
@@ -155,6 +162,11 @@ func (t ConvertToProtobuf) QueryDataResponse(res *QueryDataResponse) (*pluginv2.
 		}
 		if dr.Error != nil {
 			pDR.Error = dr.Error.Error()
+			if dr.ErrorDetails == nil {
+				pDR.ErrorDetails = &pluginv2.ErrorDetails{Status: int32(calculateErrorStatus(dr.Error))}
+			} else {
+				pDR.ErrorDetails = t.ErrorDetails(dr.ErrorDetails)
+			}
```

Comment:
This will not apply for core plugins. Guess that's fine, but for core plugins we might want to wrap [this](https://github.com/grafana/grafana/blob/f2d8b1ceee5f6beffb4903a8968b176aebd24730/pkg/plugins/backendplugin/coreplugin/registry.go#L102) in a QueryDataHandler that sets the ErrorDetails which might imply that calculateErrorStatus should be a public function?

Reply:
I think that wrapping down that low would be a bit tricky to find and reason about. I've just exported the function and used it where I think to expect such logic https://github.com/grafana/grafana/pull/49709/files#diff-da686bdc3301bd37626aec395407c56a06696a3a9613499bcbe9f80078d73079R74-R78